src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp

```diff
@@ -3945,6 +3950,13 @@ namespace OpenMS::Internal
                 filter.push(os);
                 output_stream = &filter;
             }
+            else if (options_.getWriteIndex())
+            {
+                // For non-compressed output with indexing, use a counter to track positions
+                filter.push(counter_filter);
+                filter.push(os);
+                output_stream = &filter;
```

Comment:
this needs some benchmarking, since `tellp()` may be much faster than the counter_filter.
If that is the case, we can discuss options.